### PR TITLE
replacement of "request.getRemoteHost()" by "WebUtils.getRemoteAddr()"

### PR DESCRIPTION
**Pull request by zhengyanbin**
help our‘s web server behind the proxy server get the true client ip

### CODE REVIEW

File: web/src/main/java/org/apache/shiro/web/util/WebUtils.java [hunk 1]
```diff
@@ -92,6 +92,32 @@
      */
     public static final String DEFAULT_CHARACTER_ENCODING = "ISO-8859-1";
 
+    public static String getRemoteAddr(ServletRequest servletRequest) {
+        HttpServletRequest request = toHttp(servletRequest);
+        if (request == null) {
+            return "unknown";
+        }
+        String ip = request.getHeader("x-forwarded-for");
```
Comment (bdemers) [inline web/src/main/java/org/apache/shiro/web/util/WebUtils.java]: Why is `x-forwarded-for` used twice?

File: web/src/main/java/org/apache/shiro/web/util/WebUtils.java [hunk 2]
```diff
@@ -92,6 +92,32 @@
...
+        if (ip == null || ip.length() == 0 || "unknown".equalsIgnoreCase(ip)) {
+            ip = request.getHeader("Proxy-Client-IP");
+        }
+        if (ip == null || ip.length() == 0 || "unknown".equalsIgnoreCase(ip)) {
+            ip = request.getHeader("X-Forwarded-For");
```
Comment (bdemers) [inline web/src/main/java/org/apache/shiro/web/util/WebUtils.java]: X-Forwarded-For could be a list:
https://en.wikipedia.org/wiki/X-Forwarded-For#Format

File: web/src/main/java/org/apache/shiro/web/util/WebUtils.java [hunk 3]
```diff
@@ -92,6 +92,32 @@
...
+        }
+        if (ip == null || ip.length() == 0 || "unknown".equalsIgnoreCase(ip)) {
+            ip = request.getHeader("WL-Proxy-Client-IP");
+        }
+        if (ip == null || ip.length() == 0 || "unknown".equalsIgnoreCase(ip)) {
+            ip = request.getHeader("X-Real-IP");
+        }
+
+        if (ip == null || ip.length() == 0 || "unknown".equalsIgnoreCase(ip)) {
```
Comment (bdemers) [inline web/src/main/java/org/apache/shiro/web/util/WebUtils.java]: why is `"unknown"` used?

**Comment by fpapon** [comment]
@zhengyanbin Can you fix the comments please? We can add this PR into the next release.

**Pull request closed without merging**